### DISCUSSION TITLE
typed-document-node requires documentNode (doesn't support gql tag) - why?

### DISCUSSION THREAD

**Discussion by CarsonF** [dotansimha/graphql-code-generator#4512] [General] [answered]
I'm wondering why typed-document-node requires `documentMode: documentNode` instead of allowing gql tags.
I know these are generated files but I'd like to keep them smaller since they are committed and leave that transformation to babel/webpack.

This looks to be working (written manually)
```tsx
import { gql } from 'graphql-tag';
import type { TypedDocumentNode as DocumentNode } from '@graphql-typed-document-node/core';

export const SessionDocument: DocumentNode<
  SessionQuery,
  SessionQueryVariables
> = gql`
  query Session {
    // omitted
  }
`;

**Comment by dotansimha** [comment, marked as answer]
@CarsonF You are right, no reason to force that. That's actually a bug, and not intentional. 

I fixed it in: https://github.com/dotansimha/graphql-code-generator/pull/4518 , you can find an alpha in the PR comments, and stable will be available later today.

**Reply by CarsonF** [reply]
Amazing, thanks!

**Comment by somtor** [comment]
สามรถได้เลยไหม
